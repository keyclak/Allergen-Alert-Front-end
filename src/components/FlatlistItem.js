import React, {useState} from 'react'
import { View, Text, StyleSheet, Pressable, Image} from 'react-native';
import { ListItem, Icon } from 'react-native-elements'
import { StyleConstants, Styles, Colors } from '../style';

export default function FlatlistItem(props) {

    function handlePress(location) {
        console.log(location)
    }

    return(
        <ListItem key={props.i} bottomDivider onPress={() => {handlePress(props.item.navTo)}}>
            <Icon name={props.item.icon} color={"#666"}/>
            <ListItem.Content>
            <ListItem.Title style={[Styles.labelText, {color:"#666"}]}>{props.item.title}</ListItem.Title>
            </ListItem.Content>
            <ListItem.Chevron color="#666"/>
        </ListItem>
    )
}