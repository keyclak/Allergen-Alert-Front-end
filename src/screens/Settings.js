import React, { useContext, useEffect, useState } from 'react';
import { View, Text, Pressable, StyleSheet} from 'react-native';
import LoadingButton from '../components/LoadingButton';
import TextLoadingButton from '../components/TextLoadingButton';
import { Styles } from '../style';
import { useLogin } from '../hooks/api';
import { AuthContext } from '../context';
import { ListItem, Icon } from 'react-native-elements'

export default function Settings() {
    const context = useContext(AuthContext);

    return (
        <View style={[Styles.container, {backgroundColor: 'white'}]}>   
            {/* <TextLoadingButton text="Log Out" onPress={() => context.logOut()} /> */}
            <ListItem style = {{width: '100%'}} bottomDivider onPress={() => context.logOut()}>
                <Icon name='keyboard-tab' color={"#666"}/>
                <ListItem.Title style={{color:"#666"}}>
                    Log Out
                </ListItem.Title>
            </ListItem>
        </View>
    );
}