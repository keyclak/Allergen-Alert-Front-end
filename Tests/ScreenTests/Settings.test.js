import React from 'react';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import FlatlistItem from '../../src/components/FlatlistItem';
import { create } from "react-test-renderer";
import { FlatList } from 'react-native-gesture-handler';
import { useNavigation } from '@react-navigation/native';
import renderer from 'react-test-renderer';
import Settings from '../../src/screens/Settings';

describe("Settings Tests", () => {
    test("Renders Correctly", () => {
        const tree = renderer.create(<Settings></Settings>);
        expect(tree.toJSON()).toMatchSnapshot();
    });
});