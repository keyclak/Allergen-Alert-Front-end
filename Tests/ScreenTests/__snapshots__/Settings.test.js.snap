// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Tests Renders Correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
      },
      Object {
        "paddingTop": 0,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderBottomWidth": 0.5,
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="padView"
    >
      <View
        style={
          Object {
            "overflow": "hidden",
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "paddingLeft": 16,
          }
        }
      />
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": undefined,
            "fontSize": 17,
          }
        }
        testID="listItemTitle"
      >
        Log Out
      </Text>
    </View>
  </View>
</View>
`;

exports[`test test test 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
      },
      Object {
        "paddingTop": 0,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderBottomWidth": 0.5,
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="padView"
    >
      <View
        style={
          Object {
            "overflow": "hidden",
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "paddingLeft": 16,
          }
        }
      />
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": undefined,
            "fontSize": 17,
          }
        }
        testID="listItemTitle"
      >
        Log Out
      </Text>
    </View>
  </View>
</View>
`;
